### ISSUE TITLE
What is the appropriate way to handle the errors returned from IDTokenVerifier.Verify()?

### ISSUE THREAD

**Issue by AyushSenapati** [coreos/go-oidc#327]
Well to give some context, I am using go-oidc in a middleware to verify JWT token. *oidc.IDTokenVerifier.Verify() returns IDToken and error. The verify function could possibly return different errors. Depending on the errors I want to return appropriate HTTP status code to the user. But it looks like the pkg does not expose error types. 

In the source code I see the errors are returned like below:
`fmt.Errorf("oidc: malformed jwt: %v", err) or fmt.Errorf("oidc: source does not exist")` etc.

So I am not sure how to compare these errors and return appropriate HTTP status code.
For now I am using `strings.Contain()` to compare, but I don't know if there is any better way to do this.

**Comment by mitar** [comment]
Yes, those errors should be made into exported values so you can use `errors.Is` to compare them.

**Comment by ericchiang** [comment]
What kind of errors would you want to differentiate? My experience with auth code is that it's best to be conservative, and effectively treat all errors as a 401.

**Comment by paddycarver** [comment]
I found this when looking into whether an issue already existed for what I wanted. My problem is that Verify can return two classes of errors: the token is invalid (can't parse the token, for example, or invalid signature, or expired, or whatever) that are fundamentally user errors, or that we don't know if the token is invalid or not because we e.g. couldn't fetch the keys from the remote key server. I'd like to be able to log one (or report it in metrics, or whatever) and have the other reported as an error, because the first class of error indicates a bad request, but the second class of error indicates all requests are likely to be rejected and I should do something about that. I'd also make the argument that the second class of error should trigger an internal server error instead of a 401.

Unless I'm misreading the code, and errors from Verify should _always_ be treated as indicating a known-invalid token, and if so, that would be good to know!
